saleor/warehouse/models.py

```diff
@@ -56,6 +56,9 @@ def applicable_for_click_and_collect(
         a single warehouse.
         """
 
+        if all(line.variant.is_preorder_active() for line in lines_qs):
```

Comment:
Not sure if this is the best way for doing it from the performance perspective as you're fetching a qs, and calling action on variant assigned to each line. Maybe we can perform such action directly on qs instead of each line one by one.